File: datasets/Python/Claude/Algorithm/easy4.py
def bigSorting(unsorted):
    """
    Sort array of numeric strings in ascending order of their integer values.
    
    Key insight: For numeric strings without leading zeros:
    1. Shorter strings represent smaller numbers
    2. For strings of same length, lexicographic order = numeric order
    """
    def compare_strings(s1, s2):
        # First compare by length
        if len(s1) != len(s2):
            return len(s1) - len(s2)
        # If same length, compare lexicographically
        if s1 < s2:
            return -1
        elif s1 > s2:
            return 1
        else:
            return 0
    
    # Using Python's built-in sort with custom key
    # Sort by (length, string_value)
    return sorted(unsorted, key=lambda x: (len(x), x))

def main():
    print("Enter the number of strings:")
    n = int(input())
    
    print(f"Enter {n} numeric strings:")
    unsorted = []
    for i in range(n):
        unsorted.append(input().strip())
    
    result = bigSorting(unsorted)
    
    print("\nSorted array:")
    for num in result:
        print(num)

if __name__ == "__main__":
    main()

File: datasets/C++/Claude/Algorithm/easy4.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

bool compareStrings(const string& s1, const string& s2) {
    // First compare by length
    if (s1.length() != s2.length()) {
        return s1.length() < s2.length();
    }
    // If same length, compare lexicographically
    return s1 < s2;
}

vector<string> bigSorting(vector<string>& unsorted) {
    sort(unsorted.begin(), unsorted.end(), compareStrings);
    return unsorted;
}

int main() {
    int n;
    cout << "Enter the number of strings: ";
    cin >> n;
    
    vector<string> unsorted(n);
    cout << "Enter " << n << " numeric strings:" << endl;
    
    for (int i = 0; i < n; i++) {
        cin >> unsorted[i];
    }
    
    vector<string> result = bigSorting(unsorted);
    
    cout << "\nSorted array:" << endl;
    for (const string& num : result) {
        cout << num << endl;
    }
    
    return 0;
}